Modules/IO/NIFTI/src/itkNiftiImageIO.cxx

```diff
@@ -994,6 +994,23 @@ NiftiImageIO::SetImageIOMetadataFromNIfTI()
   std::ostringstream intent_name;
   intent_name << nim->intent_name;
   EncapsulateMetaData<std::string>(thisDic, "intent_name", intent_name.str());
+
+  // The below since ITK 5.3
+  std::ostringstream qfac;
+  qfac << nim->qfac;
+  EncapsulateMetaData<std::string>(thisDic, "qfac", qfac.str());
```

Comment:
If you would want to have lossless conversion of those float values to string, you might consider using `itk::NumberToString`, instead of `std::ostringstream`, for example:

    EncapsulateMetaData<std::string>(thisDic, "qfac",  NumberToString<float>{}(qfac));


Just my 2 cents, Niels

Reply:
Lossless would indeed be better.  Searching the codebase, I see things like `itk::EncapsulateMetaData<double>(metaDict, "TestDouble", metaDataDouble);` elsewhere... I was mostly copying nearby code, I wonder why itkNiftiImageIO is only using strings...?

Reply:
@seanm I understand if you'd rather not directly change the way NiftiImageIO currently does float-to-string, if your intention is just to add some extra data. No problem, maybe lossless float-to-string conversion could still be added somewhere in the future.  😃

Reply:
I'm just questioning why it needs to be a string at all.  Seems `itk::EncapsulateMetaData` allows other types too... Then again, everything else is already a string, so probably best to just follow the existing pattern.

Reply:
@blowekamp might have some suggestions about metadata encapsulation.

Reply:
@seanm I think it should be converted to a number if it is a number.  I don't know why it would have been converted to a string.  I also prefer `NumberToString{}(qfac)` as the most lossless option.

I am making an issue to track the necessary update to make these conversions lossless in the meta data storage.

Reply:
So I just tried locally, and this works:

```
  EncapsulateMetaData<float>(thisDic, "qfac", nim->qfac);
```

but this does not:

```
  std::vector<float> qto_xyz;
  qto_xyz.push_back(nim->qto_xyz.m[0][0]);
  ...
  qto_xyz.push_back(nim->qto_xyz.m[3][3]);
  EncapsulateMetaData<std::vector>(thisDic, "qto_xyz", qto_xyz);
```

How should we store a 4x4 matrix of floats?  It does not seem to like std::vector.

Reply:
Judging by [this](https://github.com/InsightSoftwareConsortium/ITK/blob/v5.3rc03/Modules/Core/Common/include/itkMetaDataObject.h#L53-L54), `std::vector` should be supported. But you probably need to give it's element type too `std::vector<float>`, like this: `EncapsulateMetaData<std::vector<float>>(thisDic, "qto_xyz", qto_xyz);`

Reply:
Oh of course!  That works.  Can you tell C++ is not my mother tongue? :)

Reply:
What is the type of `qto_xyz`? Maybe you could encapsulate it directly.

Reply:
It's a niftilib struct, so not really appropriate to expose as such.